service/src/main/kotlin/no/nav/su/se/bakover/service/revurdering/RevurderingServiceImpl.kt

```diff
@@ -1242,4 +1230,142 @@ internal class RevurderingServiceImpl(
+
+        val avsluttetRevurdering = when (revurdering) {
+            is GjenopptaYtelseRevurdering -> revurdering.avslutt(begrunnelse, LocalDate.now(clock))
+            is Revurdering -> revurdering.avslutt(begrunnelse, fritekst, LocalDate.now(clock))
+            is StansAvYtelseRevurdering -> revurdering.avslutt(begrunnelse, LocalDate.now(clock))
+        }
+
+        return avsluttetRevurdering.mapLeft {
+            when (it) {
+                is GjenopptaYtelseRevurdering.KunneIkkeLageAvsluttetGjenopptaAvYtelse -> KunneIkkeAvslutteRevurdering.KunneIkkeLageAvsluttetGjenopptaAvYtelse(
+                    it,
+                )
+                is StansAvYtelseRevurdering.KunneIkkeLageAvsluttetStansAvYtelse -> KunneIkkeAvslutteRevurdering.KunneIkkeLageAvsluttetStansAvYtelse(
+                    it,
+                )
+                is KunneIkkeLageAvsluttetRevurdering -> KunneIkkeAvslutteRevurdering.KunneIkkeLageAvsluttetRevurdering(
+                    it,
+                )
+                else -> throw IllegalStateException("Det ble gitt ut en ukjent feil ved avslutting av en revurdering. feilen er $it")
+            }
+        }.map { avsluttedeRevurdering ->
+
+            if (avsluttedeRevurdering is Revurdering && avsluttedeRevurdering.forhåndsvarsel is Forhåndsvarsel.SkalForhåndsvarsles) {
+                val brev = LagBrevRequestForAvsluttingAvRevurdering(
+                    avsluttedeRevurdering,
+                    fritekst,
+                ).getOrElse {
+                    return KunneIkkeAvslutteRevurdering.FantIkkePersonEllerSaksbehandlerNavn.left()
+                }
+
+                brev.tilDokument {
+                    brevService.lagBrev(it).mapLeft { LagBrevRequest.KunneIkkeGenererePdf }
+                }.mapLeft {
+                    return KunneIkkeAvslutteRevurdering.KunneIkkeLageDokument.left()
+                }.map { dokument ->
+                    val dokumentMedMetaData = dokument.leggTilMetadata(
+                        metadata = Dokument.Metadata(
+                            sakId = revurdering.sakId,
+                            vedtakId = revurdering.tilRevurdering.id,
+                            revurderingId = revurdering.id,
+                            bestillBrev = true,
+                        ),
+                    )
+                    sessionFactory.withTransactionContext {
+                        brevService.lagreDokument(dokumentMedMetaData)
+                        revurderingRepo.lagre(avsluttedeRevurdering)
+                    }
+                }
+                avsluttedeRevurdering
+            } else {
+                revurderingRepo.lagre(avsluttedeRevurdering)
+                avsluttedeRevurdering
+            }
+        }
+    }
+
+    override fun brevutkastForAvslutting(
+        revurderingId: UUID,
+        fritekst: String?,
+    ): Either<KunneIKkeLageBrevutkastForAvsluttingAvRevurdering, Pair<Fnr, ByteArray>> {
```

Comment:
I hvilke tilfeller trenger frontend fødselsnummer her (som den ikke har fra før)?

Reply:
I routen so har vi audit log som trenger fnr. denne er for å bare brukes der